httplib.h

```diff
@@ -4030,6 +4030,7 @@ bool read_content(Stream &strm, T &x, size_t payload_max_length, int &status,
             skip_content_with_length(strm, len);
             ret = false;
           } else if (len > 0) {
+            x.body.reserve(len);
```

Comment:
CI is failing probably because there is no bound checking here.
I believe the following would be better:
```suggestion
            if (len > 0 && len < payload_max_length) {
              x.body.reserve(len);
            }
```